doc/reference/networking/can_api.rst

```diff
@@ -0,0 +1,282 @@
+Start Of Frame bit. After that, the identifiers follow. This phase is called the
+arbitration phase. During the arbitration phase, write collisions are allowed.
+They resolve by the fact that dominant bits override recessive bits.
+Nodes monitor the bus and notice their transmission being overridden
+abort their transmission. It results in the fact that lower number identifiers
+have priority over higher number identifiers.
+
+Filters are used to whitelist identifiers that are of interest for the specific
+node. An identifier that doesn't match any filter is ignored.
+Filters can either match exactly or a specified part of the identifier.
+This method is called masking.
+As an example, a mask with 11 bits set for standard or 29 bits set for extended
+identifiers must match perfectly.
+Bits that are set to zero in the mask don't care for the matching.
+The filters are implemented in hardware for the most CAN controllers.
+The hardware therefor limits the number of filters.
+Additionally, the number of filters is limited in Kconfig, to save memory.
+
+
+`CAN Wikipedia article <https://en.wikipedia.org/wiki/CAN_bus>`_.
+
+Zephyr supports following CAN features:
+
+* Standard and Extended Identifers
+* Filters with Masking
+* Loopback and Silent mode
+* Remote Request
+
+Sending
+*******
+
+The following code snippets show how to send data.
+
+.. code-block:: C
+
+  struct zcan_frame frame = {
+          .id_type = CAN_STANDARD_IDENTIFIER,
+          .rtr = CAN_DATAFRAME,
+          .ext_id = 0x123,
+          .dlc = 8
+  };
+  struct device *can_dev;
+  int ret;
+
+  frame.data = {1,2,3,4,5,6,7,8};
+
+  can_dev = device_get_binding("CAN_0");
+
+  ret = can_send(can_dev, &frame, K_MSEC(100), NULL, NULL);
+  if (ret != CAN_TX_OK) {
+          LOG_ERR("Sending failed [%d]", ret);
+  }
+
+This basic sample sends a CAN frame with standard identifier 0x123 and eight
+bytes of data. When passing NULL as the callback, as shown in this example,
+the send function blocks until the frame is sent and acknowledged by at least
+one other node or an error occurred. The timeout only takes effect on acquiring
+a mailbox. When a transmitting mailbox is assigned, sending cannot be canceled.
+
+.. code-block:: C
+
+  void tx_irq_callback(u32_t error_flags, void *arg)
+  {
+          char *sender = (char *)arg;
+
+          if (error_flags) {
+                  LOG_ERR("Sendig failed [%d]\nSender: %s\n", error_flags, sender);
+          }
+  }
+
+  int send_function(struct device *can_dev)
+  {
+          struct zcan_frame frame = {
+                  .id_type = CAN_EXTENDED_IDENTIFIER,
+                  .rtr = CAN_DATAFRAME,
+                  .ext_id = 0x1234567,
+                  .dlc = 2
+          };
+
+          frame.data = {1,2};
+          return can_send(can_dev, &frame, K_FOREVER, tx_irq_callback, "Sender 1");
+  }
+
+This example shows how to send a frame with extended identifier 0x1234567 and
```

Comment:
Same here. I'd recommend putting the description before the sample code (instead of after).